### ISSUE TITLE
vimtex's indentation does not work for begingroup/endgroup pair

### ISSUE THREAD

**Issue by krishnakumarg1984** [lervag/vimtex#1232]
In contrast to what `latexindent.pl` does, vimtex does not seem to indent `begingroup`/`endgroup` pairs.

Furthermore, the insert-mode "environment" closure `]]` does not work  (probably because vimtex is looking for the fixed pattern `\begin{env_name}`.  

Given these corner cases, I think, it is better to outsource these functionality to external tools. What do you say?

**Comment by lervag** [comment]
You really should give concrete examples. I don't quite understand you. You are talking about two different things: Indent and `]]`, and use them as an argument for using `latexindent.pl` instead of the current vimtex indentation function?

I don't mind if people want to use an external tool, of course. And if I knew of latexindent.pl when I started out, perhaps I would have relied on it from the start, but I think vimtex indentation is quite good and should work in most cases.

**Comment by krishnakumarg1984** [comment]
Yes. Understood. Give me a week or so. Thesis submission by next weekend!

**Comment by andymass** [comment]
The first part could probably be handled with ` g:vimtex_indent_conditionals`, see https://github.com/lervag/vimtex/issues/1078
